WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -122,5 +128,81 @@ public boolean onOptionsItemSelected(final MenuItem item) {
     public void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
         outState.putSerializable(WordPress.SITE, mSite);
+        if (!TextUtils.isEmpty(mCurrentSearch)) {
+            outState.putString(EXTRA_SEARCH_TERM, mCurrentSearch);
+        }
+    }
+
+    @Override
+    public void setTitle(CharSequence title) {
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setTitle(title);
+        } else {
+            super.setTitle(title);
+        }
+    }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
```

Comment:
Thanks for testing that! There's no `mPostList.filterPosts(query);` call here, that's why the search doesn't go through.